src/test/mod.rs

```diff
@@ -99,7 +98,8 @@ pub(crate) struct TestEnvironment {
     db: OnceCell<TestDatabase>,
     storage: OnceCell<Arc<Storage>>,
     frontend: OnceCell<TestFrontend>,
-    s3: OnceCell<TestS3>,
+    s3: OnceCell<Arc<Storage>>,
+    storage_db: OnceCell<Arc<Storage>>,
```

Comment:
Is there a reason to have everything inside of the test environment `Arc`'d instead of simply having the environment itself be `Arc`'d?

Reply:
Yes: having each component in its own `Arc` allows cloning it and passing it around without carrying the whole context, especially when only one of them is present. Also, currently the contextes of `cratesfyi` and the test suite are different, so we couldn't do that anyway.